### DISCUSSION TITLE
New default ruleset / `@auto`

### DISCUSSION THREAD

**Discussion by keradus** [PHP-CS-Fixer/PHP-CS-Fixer#7689] [Feature ideas]
Right now we have `@PSR-CS` / `@PER12` as default.
I would suggest a new one, like `@auto`, that would check composer.json and expand into:
```
[
    '@PHP{xy}Migration' => true,
    '@PHPUnit{xy}Migration' => true,
    '@PER-CS' => true,
]
```

(if risky fixers allowed, auto-add `:risky` version of sets)

And then, when someone would like to configure their project in details, they can build on top, eg:
```
[
    '@auto' => true,
    '@Symfony' => true,
]

**Comment by mvorisek** [comment]
`@PHPAutoMigration` and `@PHPUnitAutoMigration` would be my preferred choise.

**Reply by keradus** [reply]
I believe it's valid to add them, but I would still go with `@auto` on top. This could prevent a need to use config file / cli params for base usage.

**Comment by Wirone** [comment]
In this context it would be even better to have #7264 implemented, so final configuration can be described and it would be clear that some of the rules come from "auto" rules. In terms of the idea itself: looks OK. If I understand correctly it would determine PHP version (minimal supported?) and PHPUnit (minimal required?) only?

**Reply by keradus** [reply]
> If I understand correctly it would determine PHP version (minimal supported?) and PHPUnit (minimal required?) **only** ?

those 2 I had in mind. you have any other in mind ?

**Comment by keradus** [comment]
done in #9090
